ganga/GangaCore/Lib/Executable/Executable.py

```diff
@@ -54,7 +54,7 @@ class Executable(IPrepareApp):
     _schema = Schema(Version(2, 0), {
         'exe': SimpleItem(preparable=1, defvalue='echo', typelist=[str, File], comparable=1, doc='A path (string) or a File object specifying an executable.'),
         'args': SimpleItem(defvalue=["Hello World"], typelist=[str, File, int], sequence=1, strict_sequence=0, doc="List of arguments for the executable. Arguments may be strings, numerics or File objects."),
-        'env': SimpleItem(defvalue={}, typelist=[str], doc='Dictionary of environment variables that will be replaced in the running environment.'),
+        'env': SimpleItem(defvalue={}, typelist=[dict], doc='Dictionary of environment variables that will be replaced in the running environment.'),
```

Comment:
seems that this has to be `str` since in the schema `_check_type` method it iterates over dicts and checks the key and value are the type specified in `typelist`...see below

https://github.com/ganga-devs/ganga/blob/b9dca4b3e069d2300e50b8757b90aea7f3dff085/ganga/GangaCore/GPIDev/Schema/Schema.py#L548-L551

Reply:
Seems odd. So how do you specify the environment? Is it a string rep of a dict?

Reply:
no looking at the tests it's a dict. but just the schema checking has a special case for dicts where it checks the content against the ``typelist`

Reply:
I agree it's confusing though

Reply:
For the Dirac backend object, I notice that we simply do not have a `typelist`. https://github.com/ganga-devs/ganga/blob/b9dca4b3e069d2300e50b8757b90aea7f3dff085/ganga/GangaDirac/Lib/Backends/DiracBase.py#L112

Reply:
interesting... I'm guessing in that case it just does no type checking